src/VisualStudio/Xaml/Impl/Microsoft.VisualStudio.LanguageServices.Xaml.csproj

```diff
@@ -32,6 +32,7 @@
     <PackageReference Include="Microsoft.VisualStudio.Editor" Version="$(MicrosoftVisualStudioEditorVersion)" />
     <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="$(MicrosoftVisualStudioShell150Version)" />
     <PackageReference Include="Microsoft.VisualStudio.ComponentModelHost" Version="$(MicrosoftVisualStudioComponentModelHostVersion)" />
+    <PackageReference Include="StreamJsonRpc" Version="$(StreamJsonRpcVersion)" />
```

Comment:
Why did we have to add this here? Otherwise this picks an older version and something flags the conflict?

Reply:
so MS.VS.Editor.XXX has reference to old version and without explicit reference here, restore is fine, but msbuild complains that there are 2 different file with same name but different versions in the output dir and complains about conflicts.

I heard this is the usual pattern to fix such case from @jmarolf

Reply:
I can also confirm that these conflicts have cause me to experience an infinite amount of sadness

Reply:
Yeah, as long as we have nuget packages that do not declare their dependencies nuget will resolve everything and then msbuild will fail with ambiguities. Adding this explicit package reference will causes msbuild to consider these assemblies when generating binding redirects. @jasonmalinowski 	I assume we have bugs on the VSSDK assemblies that we need to generate nuget wrapper packages for?

Reply:
@heejaechang It was the editor package? If so, can we get a bug filed on them to fix it?

@jmarolf I'm not sure if we have bugs or not, but the packages we're building for more or less are old packages that predate us having OOP processes at all. The comment from @heejaechang about "Editor...." makes me think this is an editor bug we need them to fix.

Reply:
@jasonmalinowski we are all trying to move to newer version of StreamJsonRpc. but since 2.x and 1.x are incompatible, moving is not cheap. so we can't force them. but they know that they need to move at some point. and it is not just editor team (even if we only have the dependency on editor and platform), other teams such as settings, dataware house and etc will get the reverse problem if (editor/platform) moves because they will get the conflict since editor/platform is 2.0 and they are 1.3. so, they had to move as well.

here by editor I meant new OOP code lens API, here platform I meant HubClient in platform.

Reply:
To be clear, no one should be "forced" to upgrade just because a partner does. StreamJsonRpc 1.x and 2.0 can coexist side by side in the same appdomain. The issue you're seeing with nuget and the build would be the same even if 1.x and 2.0 were API compatible. So what you're seeing is typically any time you have transitive dependencies that happen to have compiled against different package versions.

Reply:
@heejaechang I get the big picture, I'm just wondering what happens if I delete the exact line here. Will the code still build?

Reply:
From @heejaechang's screenshots he sent me, it looks like it will build, but with warnings.

Reply:
@jasonmalinowski we get conflict warning from msbuild. @AArnott if 1.x and 2.x were compatible, it would feel much less heavy when I ask other partners to move, but this since, it require them to actually change code which requires them to test more thoroughly, I dont feel like pushing for it.